azure-pipelines.yml

```diff
@@ -0,0 +1,85 @@
+# ASP.NET
+# Build and test ASP.NET projects.
+# Add steps that publish symbols, save build artifacts, deploy, and more:
+# https://docs.microsoft.com/azure/devops/pipelines/apps/aspnet/build-aspnet-4
+
+trigger:
+  batch: true
+  branches:
+    include:
+    - '*'
+  tags:
+    include:
+    - 'v*'
+
+pr:
+- 'development'
+- 'master'
+
+pool:
+  vmImage: 'windows-latest'
+
+steps:
+# -------------------
+# Install Dependencies
+# -------------------
+- task: UseDotNet@2
+  displayName: Require .NET 2.1
+  inputs:
+    packageType: 'runtime'
+    version: '2.1.x'
+
+- task: UseDotNet@2
+  displayName: Require .NET 3.1 SDK for Build Process
+  inputs:
+    packageType: 'sdk'
+    version: '3.1.x'
+
+- task: PowerShell@2
+  displayName: Setup Required Dependencies
+  inputs:
+    filePath: 'setup.ps1'
+
+- task: PowerShell@2
+  displayName: Generate Version using GitVersion
+  inputs:
+    targetType: 'inline'
+    script: dotnet gitversion /output buildserver
+
+# -------------------
+# RUN PSAKE SCRIPT
+# -------------------
+
+- task: PowerShell@2
+  displayName: Build and Test
+  continueOnError: true
+  inputs:
+    targetType: 'inline'
+    script: |
+      invoke-psake run-ci -properties @{'version'='$(Build.BuildNumber)'}
+
+# -------------------
+# PIPELINE FOLLOW-UP
+# -------------------
+
+- task: PublishTestResults@2
+  inputs:
+    testResultsFormat: 'VSTest'
+    testResultsFiles: '$(Build.SourcesDirectory)/src/BulkWriter.Tests/TestResults/*.xml'
+  displayName: Publish Test Results to Pipeline
```

Comment:
To go with the change to running the powershell build above, add here:
`condition: succeededOrFailed()`